receiver.go

```diff
@@ -424,11 +455,17 @@ func newReceiver(source string, session *Session, opts *ReceiverOptions) (*Recei
 		},
 		autoSendFlow:  true,
 		receiverReady: make(chan struct{}, 1),
+		messagesE:     make(chan *queue.Queue[Message], 1),
+		messagesP:     make(chan *queue.Queue[Message], 1),
 		batching:      defaultLinkBatching,
 		batchMaxAge:   defaultLinkBatchMaxAge,
 		maxCredit:     defaultLinkCredit,
 	}
 
+	// messagesE MUST always be the starting channel that's populated with an empty queue.
+	// the queue's segment size is made relative to the session's incoming window.
+	r.messagesE <- queue.New[Message](int(session.incomingWindow) / 10)
```

Comment:
Is /10 significant or it's just trying to keep it reasonable compared to the window?